serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/States.kt

```diff
@@ -461,9 +464,19 @@ class MatrixChatRoom(private val msession: MatrixSession, val room_ids: List<Str
             got_messages
         }
     }
+    fun getDisplayNameForUser(sender: String) : String {
+        val (displayname, _) = msession.getDiplayNameAndAvatarFilePath(sender, room_id)
+        return displayname ?: ""
+    }
+    fun getUnformattedBody(formatted_body: String) : String {
+        val link_regex = Regex("<a href=\"https://matrix.to/#/[^:]*:[^>]*\">(.*)</a>")
+        return formatted_body.replace(link_regex,"$1")
+    }
     val window_forward_length: Int = if (message_window_base != null) { window_forward_length_in } else { 0 }
-    fun sendMessage(msg: String): MatrixState {
-        when (val sendMessageResult = msession.sendMessage(msg, room_id)) {
+    fun sendMessage(msg: String, format: String = ""): MatrixState {
```

Comment:
I guess this would make format a bool, though we could almost  remove it entirely and see if msg == getUnformattedBody(msg)